server-side-swift-app-with-vapor-and-droplet.md

```diff
@@ -0,0 +1,172 @@
+
+`curl -sL toolbox.vapor.sh | bash`
+
+To verify if the installation was successful, run `vapor --help`, which should show a list of commands. To update the toolbox, just run `vapor self update`.
+
+To make sure everything is running fine, let's build a Hello World app.
+
+### Hello World App
+
+First of all, let's create the new Vapor project.
+
+`vapor new HelloWorld`
+
+Vapor will generate a folder structure with some files. Vapor command line interface uses [Swift Package Manager](https://swift.org/package-manager/) to manage the dependencies.
+
+If you are using macOS and want to edit the project on Xcode, run `vapor xcode -y` in the HelloWorld subdirectory to generate and open an Xcode project.
+
+Look for the main.swift file and place the following code there.
+
+```swift
+import Vapor
+
+let drop = Droplet()
+
+drop.get { req in
+    return try drop.view.make("welcome", [
+    	"message": drop.localization[req.lang, "welcome", "title"]
+    ])
+}
+
+drop.resource("posts", PostController())
+
+drop.run()
+
+```
+
+> Note: on version 1.1, Vapor already generates this code when you create a new project. But it's here just in case things change, considering that Vapor is in a fast development pace.
+
+After that, run the two following commands.
+
+```
+vapor build
+vapor run serve
+```
+
+And it should be done. Now vapor is serving its HelloWorld page on localhost:8080. To check that things are working out, just open your browser and type `localhost:8080`.
+
+
+## Bulding a Contacts App
+
+### Playing with Droplet
+
+Droplet is a container that makes it easy to use many of Vapor's features. It's responsible for starting and running the server, set up environments, register routes, and more. It's necessary to import Vapor to initialize and run a droplet.
+
+```swift
+import Vapor
+
+let drop = Droplet()
+
+// config, registering routes, etc
+
+drop.run()
+```
+
+Let's now register a get request at localhost:8080/hello.
+
+```swift
+drop.get("hello") { request in
+    return "Hello World!"
+}
+```
+
+The first parameter is the endpoint, and the closure returns the response of the request. The return must be of a type who conforms to the `ResponseRepresentable` protocol. Strings and JSON already conform to this protocol, but it's also possible to create a custom response.
+
+```swift
+drop.get("hello") { request in
+	return Response(status: .ok, headers: ["Content-Type": "text/plain"], body: "Hello, World!")
+}
+```
+
+It's also pretty straightforward to create a post request and retrieve sent JSON data.
+
+```swift
+drop.post("contacts", "create") { request in
+    return try JSON(node: [
+        "response": request.data["name"]?.string
+        ])
+}
+```
+
+This register a POST HTTP endpoint at localhost:8080/contacts/create and returns a JSON response. It gets the data sent as JSON in the request for the key test and retrieves its string value on `request.data["test"]?.string`.
```

Comment:
"Which sentence is misleading? The one you mentioned in the prior comment?" <= exactly